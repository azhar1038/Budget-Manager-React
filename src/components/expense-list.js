import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './expense-list-item'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses Found</p>
            ):(
                props.expenses.map((expense)=>(
                    <ExpenseListItem
                        key={expense.id}
                        {...expense}
                    />
                ))
            )
        }
    </div>
)

const mapStateToProps = (state)=>({
    expenses: selectExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpenseList)