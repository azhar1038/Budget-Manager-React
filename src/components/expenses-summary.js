import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import totalExpenses from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = ({expenseCount, expensesTotal})=>{
    const expenseWord = expenseCount > 1? 'expenses': 'expense'
    const formattedTotal = numeral(expensesTotal/100).format('0,0.00')
    return(
        <div>
            Viewing {expenseCount} {expenseWord} totalling â‚¹{formattedTotal}
        </div>
    )
}

const mapStateToProps = (state)=>{
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: expenses.length,
        expensesTotal: totalExpenses(expenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)