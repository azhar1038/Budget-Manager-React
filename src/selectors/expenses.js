//Get Visible Expenses

const getVisibleExpenses = (expenses, {text, sortBy, startDate, endDate})=>{
    text = text.toLowerCase()
    return expenses.filter((expense)=>{
        const startDateMatch = startDate ? startDate.isSameOrBefore(expense.createdAt) : true
        const endDateMatch = endDate ? endDate.isSameOrAfter(expense.createdAt) : true
        const textMatch = expense.description.toLowerCase().includes(text)

        return startDateMatch && endDateMatch && textMatch
    }).sort((a, b)=>{
        if(sortBy === 'date'){
            return a.createdAt < b.createdAt ? 1 : -1 
        }else if(sortBy === 'amount'){
            return a.amount < b.amount ? 1 : -1
        }
    })
}

export default getVisibleExpenses