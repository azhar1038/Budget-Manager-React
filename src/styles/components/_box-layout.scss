.box-layout{
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box{
    background: fade-out($color: white, $amount: .25);
    border-radius: 4px;
    padding: $l-size $m-size;
    text-align: center;
    max-width: 25rem;
    margin: $s-size;
}

.box-layout__title{
    margin: 0 0 $m-size 0;
    line-height: 1;
}