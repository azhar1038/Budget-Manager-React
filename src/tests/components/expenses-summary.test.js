import React from 'react'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import { ExpensesSummary } from '../../components/expenses-summary'

test('Should render expense summary for 1 expense correctly', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123.90}/>)
    expect(wrapper).toMatchSnapshot()
})

test('Should render expense summary for multiple expense correctly', ()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={12367.90}/>)
    expect(wrapper).toMatchSnapshot()
})