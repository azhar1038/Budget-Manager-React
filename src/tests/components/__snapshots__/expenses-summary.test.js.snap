// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense summary for 1 expense correctly 1`] = `
<div>
  Viewing 
  1
   
  expense
   totalling ₹
  1.24
</div>
`;

exports[`Should render expense summary for multiple expense correctly 1`] = `
<div>
  Viewing 
  2
   
  expenses
   totalling ₹
  123.68
</div>
`;
